endpoints/bookstore-grpc/bookstore_client.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""The Python GRPC Bookstore Client Example."""
+
+import argparse
+import grpc
+
+from google.protobuf import empty_pb2
+
+import bookstore_pb2
+
+
+class ApiKeyMetadata(object):
+    def __init__(self, api_key):
+        self.api_key = api_key
+
+    def __call__(self, metadata):
+      if metadata:
+        for k, v in metadata:
+          yield (k, v)
+      if self.api_key:
+        yield ('x-api-key', self.api_key)
+
+
+def run(host, port, api_key, timeout):
+  """Makes a basic ListShelves call against a gRPC Bookstore server."""
+
+  channel = grpc.insecure_channel('{}:{}'.format(host,port))
+
+  stub = bookstore_pb2.BookstoreStub(channel)
+  shelves = stub.ListShelves(empty_pb2.Empty(), timeout,
```

Comment:
In general, put a newline after the opening `(` to avoid hanging indents. (please do this throughout).